packages/sdk/src/fetcher/on-chain/index.ts

```diff
@@ -63,6 +64,26 @@ class Fetcher implements IPartialCarrotFetcher {
         ).slice();
     }
 
+    public async fetchLatestKPITokenAddresses({
+        publicClient,
+        limit,
+    }: FetchLatestKpiTokenAddressesParams): Promise<Address[]> {
+        const chainId = await publicClient.getChainId();
+        enforce(chainId in ChainId, `unsupported chain with id ${chainId}`);
+        const chainAddresses = CHAIN_ADDRESSES[chainId as ChainId];
+        const finalLimit = limit || 5;
+        const toIndex = await this.fetchKPITokensAmount({ publicClient });
+        const fromIndex = toIndex - finalLimit > 0 ? toIndex - finalLimit : 0;
+        return (
+            await publicClient.readContract({
+                abi: FACTORY_ABI,
+                address: chainAddresses.factory,
+                functionName: "enumerate",
+                args: [BigInt(fromIndex), BigInt(toIndex)],
+            })
+        ).slice();
```

Comment:
I'm wondering if instead of slicing (I'm guessing this is to make the end result not readonly) we should just do an explicit casting. Because there's nothing in the code making this readonly at runtime. It's just a Typescript compile time thing. So an `as` should do the trick.